crates/loam-cli/examples/soroban/nft/src/nft.rs

```diff
@@ -0,0 +1,126 @@
+use loam_sdk::{
+    soroban_sdk::{self, contracttype, env, Address, Bytes, Lazy, Map, Vec},
+    IntoKey,
+};
+use loam_subcontract_core::Core;
+
+use crate::{
+    subcontract::{IsInitable, IsNonFungible},
+    Contract,
+};
+
+#[contracttype]
+#[derive(IntoKey)]
+pub struct MyNonFungibleToken {
+    admin: Address,
+    name: Bytes,
+    total_count: u32,
+    owners_to_nft_ids: Map<Address, Map<u32, ()>>, // the owner's collection
+    nft_ids_to_owners: Map<u32, Address>,
+    nft_ids_to_metadata: Map<u32, Bytes>,
+}
+
+impl MyNonFungibleToken {
+    #[must_use]
+    pub fn new(admin: Address, name: Bytes) -> Self {
+        MyNonFungibleToken {
+            admin,
+            name,
+            total_count: 0,
+            owners_to_nft_ids: Map::new(env()),
+            nft_ids_to_owners: Map::new(env()),
+            nft_ids_to_metadata: Map::new(env()),
+        }
+    }
+}
+
+impl Default for MyNonFungibleToken {
+    fn default() -> Self {
+        MyNonFungibleToken::new(env().current_contract_address(), Bytes::new(env()))
+    }
+}
+
+impl IsInitable for MyNonFungibleToken {
+    fn nft_init(&self, admin: Address, name: Bytes) {
+        Contract::admin_get().unwrap().require_auth();
+        MyNonFungibleToken::set_lazy(MyNonFungibleToken::new(admin, name));
+    }
+}
+
+impl IsNonFungible for MyNonFungibleToken {
+    // Mint a new NFT with the given owner address and metadata, returning the id
+    fn mint(&mut self, owner: Address, metadata: Bytes) -> u32 {
+        owner.require_auth();
```

Comment:
Our implementation allows anyone to mint themselves NFTs? Seems like most implementations would want to limit this to `Contract::admin_get()`, right?

Reply:
@willemneal I also want to make sure that I understand how this works correctly, because after I deployed an update to this contract so that `mint` has `self.admin.require_auth()` I was still able to invoke a mint tx with a non-admin as the source acct and I'm not quite sure why. 

When I invoke mint like this, it fails as expected because `molly` is not the admin account.
```
stellar contract invoke \
--id CDLCRZ6ICRW5KB7RFKKJEJ57JQHXLRS2MIBDZ6B4LAZGLKY2WLAJYYLO \
--source molly \
--network testnet \
-- \
mint \
--owner molly \
--metadata 0a2020...

error: Missing signing key for account GDVVBBT5YPLPGN6K2JTCLDSOCF467O2JPWLB2JCF4MQMX4BFOYFM7NZF (this is the admin address)
```

However, when I mint with `molly` as the source, but with `gene`, who is the admin, as the owner it works. Is the owner also signing the tx and that is why it succeeds?
```
stellar contract invoke \
--id CDLCRZ6ICRW5KB7RFKKJEJ57JQHXLRS2MIBDZ6B4LAZGLKY2WLAJYYLO \
--source molly \
--network testnet \
-- \
mint \
--owner gene \
--metadata 0a20202...

1
```

Reply:
This is a shortcoming of the current stellar-cli. It doesn't show you that it signs both the transaction with the source account and the auth entry that is required for `gene`.  If `gene` isn't a global config, then you can try from a different directory with a global key. Then the CLI won't have access to `gene`'s key and won't be able to sign the auth entry. Again this will be make more explicit in the new signing PR.

Reply:
Cool, thanks for that explanation. 🙏  That's what I was thinking, but needed a sanity check to make sure I wasn't missing something in the NFT contract implementation.